### ISSUE TITLE
Feature Request: Support Multiple Revisions

### ISSUE THREAD

**Issue by keithmattix** [cert-manager/istio-csr#86]
Due to the explicit certificate naming in the [helm chart](https://github.com/cert-manager/istio-csr/blob/master/deploy/charts/istio-csr/templates/certificate.yaml), istio-csr's helm chart doesn't support non-default revisions. 

Are there any plans to remediate this? One approach that comes to mind would be to accept an array of revisions in the helm chart (with the option to exclude the default revision) and loop over that value in the `certificate.yaml` template.

**Comment by JoshVanL** [comment]
Hi @keithmattix, could you expand a bit on what you mean by `revision`, do you mean the `revisionHistoryLimit`?

**Comment by keithmattix** [comment]
Thanks for the response @JoshVanL; sorry I should've been more clear. I'm referring to Istio [revisions](https://istio.io/latest/docs/setup/upgrade/canary/#control-plane) for use with canary upgrades.

**Comment by JoshVanL** [comment]
Thanks @keithmattix, wasn't familiar of this feature with istio but is pretty cool!

Would exposing something _like_ `app.istioRevision` in the helm chart make sense here?

/kind feature
/assign

**Comment by keithmattix** [comment]
Yes, I think something like that could work! It would probably need to be a list/array since one would generally have multiple revisions in a cluster during a canary upgrade. `app.istioRevisions` would probably suffice.

**Comment by JoshVanL** [comment]
Cool, thanks @keithmattix. I have a branch at work in the background :slightly_smiling_face: 

Will likely cover the following cases:
- Empty revision list `[]` -> `""`
- Default in the revision list gets converted `"default"` -> `""`
- Multiple revisions would produce `["default", "foo", "bar"]` -> `"", "foo", "bar"`

Do we need to account for the case where the Helm revision value has both `""` and `"default"`? This would produce a Certificate with DNS that has duplicate entries. Maybe this is fine, but can spend some time to remove duplicates in the helm template logic (:sweat_smile:).

**Comment by keithmattix** [comment]
You could probably just use the `uniq` function in Helm to filter the list before looping over it right?
